### ISSUE TITLE
ActiveRecord still used

### ISSUE THREAD

**Issue by brilnius** [CanCanCommunity/cancancan#742]
Hello everybody.

ActiveRecord is still used in code.

cancan/rule.rb:72:

```
    def with_scope?
      @conditions.is_a?(ActiveRecord::Relation)
    end
```

This causes an exception when ActiveRecord is not used in the project (e.g. using Mongoid).
The exception is raised when you inspect() any CanCan::Rule.

I couldn't advise how to address this properly. For the time being we use a monkey patch where with_scope? returns false.

PS: thank you for all the very good work.

**Comment by Liberatys** [comment]
@brilnius I believe this is my doing 🤦🏻 I'll have a look on how to best handle non activerecord events 👍🏻 Thank you for reporting

**Comment by danlo** [comment]
👍🏼 I encountered this issue and resolved with the same issue.  I also use mongoid. 

```

module CanCan
  class Rule
    def with_scope?() = nil
  end
end
```

**Comment by IJustDev** [comment]
I think this can be closed by having #766 merged.

**Comment by danlo** [comment]
I manually overloaded with the solution in #766 worked.  All my test code passes.

**Comment by Liberatys** [comment]
The extraction/separation of ActiveRecord in the codebase is on the roadmap 😉.

**Comment by coorasse** [comment]
Fixed in 3.4.0
